ui/src/components/SampleView/GridForm.jsx

```diff
@@ -135,93 +134,86 @@ export default class GridForm extends React.Component {
-        <Row className="gridform">
-          <Col xs={8}>
-            <Table striped hover responsive>
-              <thead>
-                <tr>
-                  <th>Name</th>
-                  {this.use_advanced_settings
-                    ? [<th>V-Space (µm)</th>, <th>H-Space (µm)</th>]
-                    : null}
-                  <th>Dim (µm)</th>
-                  <th>#Cells</th>
-                  <th>R x C</th>
-                  <th>&Omega;</th>
-                  <th />
-                  <th />
-                  <th />
-                </tr>
-              </thead>
-              <tbody>{this.getGridControls()}</tbody>
-            </Table>
-          </Col>
-          <Col xs={4} style={{ marginTop: '20px' }}>
-            <Form>
-              <Form.Group className="mb-2" as={Row}>
-                <Col sm="4">
-                  {' '}
-                  <Form.Label>Opacity</Form.Label>{' '}
-                </Col>
-                <Col sm="1"> : </Col>
-                <Col sm="7">
-                  <Form.Control
-                    style={{ width: '100px', padding: '0' }}
-                    className="bar"
-                    type="range"
-                    id="overlay-control"
-                    min="0"
-                    max="1"
-                    step="0.05"
-                    defaultValue={this.props.getGridOverlayOpacity()}
-                    onChange={this.props.setGridOverlayOpacity}
-                    name="overlaySlider"
-                  />
-                </Col>
-              </Form.Group>
-              <Form.Group className="mb-2" as={Row}>
-                <Col sm="4">
-                  {' '}
-                  <Form.Label>Heat map</Form.Label>{' '}
-                </Col>
-                <Col sm="1"> : </Col>
-                <Col sm="7">
-                  <Form.Check
-                    name="resultType"
-                    type="radio"
-                    onClick={() => this.props.setGridResultType('heatmap')}
-                    checked={this.props.gridResultType === 'heatmap'}
-                  />
-                </Col>
-              </Form.Group>
-              <Form.Group as={Row}>
-                <Col sm="4">
-                  {' '}
-                  <Form.Label>Crystal map</Form.Label>{' '}
-                </Col>
-                <Col sm="1"> : </Col>
-                <Col sm="7">
-                  <Form.Check
-                    name="resultType"
-                    type="radio"
-                    onClick={() => this.props.setGridResultType('crystalmap')}
-                    checked={this.props.gridResultType === 'crystalmap'}
-                  />
-                </Col>
-              </Form.Group>
-            </Form>
-          </Col>
-        </Row>
-      </Draggable>
-    );
-
-    return this.props.show ? gridForm : null;
-  }
+  return show ? (
+    <Draggable
+      position={position}
```

Comment:
If you can keep the component [uncontrolled](https://react.dev/learn/sharing-state-between-components#controlled-and-uncontrolled-components) using `defaultPosition`, it will save a lot of re-renders when dragging:

```tsx
defaultPosition={position}
```

Reply:
Actually, never mind. Everything will re-render anyway because of the local state. You could store the position in a ref to really avoid re-renders, but it's not worth it.